app/src/main/java/org/jellyfin/mobile/media/MediaService.kt

```diff
@@ -529,28 +309,122 @@ class MediaService : MediaBrowserServiceCompat() {
             extras: Bundle?,
             cb: ResultReceiver?
         ): Boolean = false
+    }
 
-        fun createMediaItem(mediaId: String): ExoPlayerMediaItem {
-            val url = "${apiClient.serverAddress}/Audio/${mediaId}/universal?" +
-                "UserId=${apiClient.currentUserId}&" +
-                "DeviceId=${URLEncoder.encode(apiClient.deviceId, Charsets.UTF_8.name())}&" +
-                "MaxStreamingBitrate=140000000&" +
-                "Container=opus,mp3|mp3,aac,m4a,m4b|aac,flac,webma,webm,wav,ogg&" +
-                "TranscodingContainer=ts&" +
-                "TranscodingProtocol=hls&" +
-                "AudioCodec=aac&" +
-                "api_key=${apiClient.accessToken}&" +
-                "PlaySessionId=${UUID.randomUUID()}&" +
-                "EnableRemoteMedia=true"
-
-            return ExoPlayerMediaItem.fromUri(url)
+    /**
+     * Listen for notification events.
+     */
+    private inner class PlayerNotificationListener : PlayerNotificationManager.NotificationListener {
+        override fun onNotificationPosted(notificationId: Int, notification: Notification, ongoing: Boolean) {
+            if (ongoing && !isForegroundService) {
+                val serviceIntent = Intent(applicationContext, this@MediaService.javaClass)
+                ContextCompat.startForegroundService(applicationContext, serviceIntent)
+
+                startForeground(notificationId, notification)
+                isForegroundService = true
+            }
+        }
+
+        override fun onNotificationCancelled(notificationId: Int, dismissedByUser: Boolean) {
+            stopForeground(true)
+            isForegroundService = false
+            stopSelf()
+        }
+    }
+
+    /**
+     * Listen for events from ExoPlayer.
+     */
+    private inner class PlayerEventListener : Player.EventListener {
+        override fun onPlayerStateChanged(playWhenReady: Boolean, playbackState: Int) {
+            when (playbackState) {
+                Player.STATE_BUFFERING,
+                Player.STATE_READY -> {
+                    notificationManager.showNotificationForPlayer(currentPlayer)
+                    if (playbackState == Player.STATE_READY) {
+                        // TODO: When playing/paused save the current media item in persistent storage
```

Comment:
TODO for now or future?

Reply:
Definitely future. Need a database for that first.